worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/implementation/chunk/ChunkHolder.java

```diff
@@ -79,8 +78,13 @@ public synchronized IBlockDelegate getDelegate() {
      * If the chunk is currently being "called", this method will block until completed.
      */
     private void checkAndWaitOnCalledLock() {
-        if (calledLock.tryLock()) {
-            calledLock.unlock();
+        if (calledLock.isLocked()) {
+            synchronized (this) {
+                if (calledLock.isLocked()) {
+                    calledLock.lock();
+                    calledLock.unlock();
+                }
+            }
```

Comment:
can't we just do lock unlock here?

Reply:
yeah not sure why I tried to have some sort of optimisation here. I'll leave the first `isLocked` call there though as it's faster than locking and unlocking, and `checkAndWaitOnCalledLock` is called frequently